Scripts/RemediationScripts/Remediate-SetSQLServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,886 @@
+        Write-Host $([Constants]::SingleDashLine)
+        return
+        }
+        Write-Host "Fetching all SQL Servers failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No SQL Server(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }  
+        $validResources | ForEach-Object { 
+            try
+            {
+            $name = $_.ResourceId.Split('/')[8]
+            $resourceGroupName = $_.ResourceId.Split('/')[4]
+            $resSqlServer = Get-AzSqlServer  -Name $name -ResourceGroupName $resourceGroupName  -ErrorAction SilentlyContinue
+            $sqlServerResources = $sqlServerResources + $resSqlServer
+            }
+            catch
+            {
+            Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host "Skipping the Resource: [$($_.ResourceName)]..."
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+            $logResource.Add("ResourceName",($_.ResourceName))
+            $logResource.Add("Reason","Valid resource id(s) not found in input json file.")    
+            $logSkippedResources += $logResource
+            Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+    }
+    else
+    {
+        # No file path provided as input to the script. Fetch all SQL Servers in the Subscription.
+        if ([String]::IsNullOrWhiteSpace($FilePath))
+        {
+            Write-Host "`nFetching all SQL Servers in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+            # Get all SQL Servers in the Subscription
+            $sqlServerResources = Get-AzSqlServer  -ErrorAction Stop
+
+       
+            $totalsqlServerResources = ($sqlServerResources | Measure-Object).Count
+        
+        }
+        else
+        {
+                if (-not (Test-Path -Path $FilePath))
+                {
+                    Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
```

Comment:
As script might terminate here, use double dash line after this statement,

Reply:
added